src/taskFunctions.js

```diff
@@ -0,0 +1,98 @@
+import chkboxStatus from './chkbox.js';
+
+class Task {
+  constructor(i, c = false, d) {
+    this.index = i;
+    this.completed = c;
+    this.description = d;
+  }
+}
+
+let taskList = [];
+
+const saveStorage = () => {
+  localStorage.setItem('tasks', JSON.stringify(taskList));
+};
+
+const addLi = (ul, index, completed, description) => {
+  const item = document.createElement('li');
+  const cBox = document.createElement('input');
+  cBox.type = 'checkbox';
+  cBox.id = index;
+  cBox.checked = completed;
+  cBox.addEventListener('change', () => {
+    if (chkboxStatus(cBox)) {
+      taskList[parseInt(cBox.id, 10)].completed = true;
+    } else {
+      taskList[parseInt(cBox.id, 10)].completed = false;
+    }
+    saveStorage();
+  });
+  const label = document.createElement('label');
+  label.contentEditable = true;
+  label.innerText = description;
+  label.addEventListener('focus', () => {
+    label.parentNode.style.backgroundColor = 'lightyellow';
+    label.parentNode.childNodes[2].style.pointerEvents = 'initial';
+    label.parentNode.childNodes[2].childNodes[0].style.display = 'none';
+    label.parentNode.childNodes[2].childNodes[2].style.display = 'initial';
+  });
+  label.addEventListener('focusout', (e) => {
+    label.parentNode.style.backgroundColor = 'initial';
+    label.parentNode.childNodes[2].childNodes[0].style.display = 'initial';
+    label.parentNode.childNodes[2].childNodes[2].style.display = 'none';
+    taskList[parseInt(cBox.id, 10)].description = e.composedPath()[0].innerText;
+    saveStorage();
+    setTimeout(() => {
+      label.parentNode.childNodes[2].style.pointerEvents = 'none';
+    }, 500);
+  });
+  const elBtn = document.createElement('button');
```

Comment:
- [ ]  following the DRY principles, is recommended to use only one innerHTML instead of all the create element and appendchild a lot of times, for example:

````
const ulist = document.queryselector("#myDiv");
ulist.innerHTML +=
`
  <div class="myDivClass">
    <li class="myclas">mi item is ${myvariable}</li>
  </div>
`;
````

Reply:
I know your code is an example but nesting the `<li>` with a `<div>` can be problematic.
But I think I didn't understand your recommendation here since I need to add EventListener to every element in the list and I don't know a way to do that with the `innerHTML` tag.